app/src/androidTest/java/com/github/sdp/mediato/NewItemFragmentTest.java

```diff
@@ -0,0 +1,176 @@
+            FragmentManager fragmentManager = activity.getSupportFragmentManager();
+            NewItemFragment newItemFragment = new NewItemFragment();
+
+            // Pass the username to the fragment like at profile creation
+            Bundle bundle = new Bundle();
+
+            bundle.putSerializable("media", review.getMedia());
+            newItemFragment.setArguments(bundle);
+            fragmentManager.beginTransaction().replace(R.id.main_container, newItemFragment)
+                    .commitAllowingStateLoss();
+        });
+    }
+
+    // check if all the film info is displayed
+    @Test
+    public void checkFragmentIsLaunchedWithCorrectInfo() {
+        onView(withId(R.id.item_title)).check(matches(withText(review.getMedia().getTitle())));
+        onView(withId(R.id.item_description_text)).check(matches(withText(review.getMedia().getSummary())));
+    }
+
+
+    // Check if indicator text is well displayed when using the slide bar
+    @Test
+    public void checkSlideBarAndIndicator() {
+        seekBar.perform(setProgress(5));
+        seekBarIndicator.check(matches(withText("5")));
+    }
+
+    // Test that no error message is displayed when review is correct
+    @Test
+    public void checkNoErrorsWhenAddingACorrectLengthComment() {
+        seekBar.perform(setProgress(review.getGrade()));
+
+        String comment = review.getComment();
+        editText.perform(typeText(
+                comment.length() >= MAX_REVIEW_LENGTH ? comment.substring(0, MAX_REVIEW_LENGTH - 1) : comment));
+        editText.perform(closeSoftKeyboard());
+
+       addItemButton.perform(click());
+
+        onView(withId(R.id.main_container))
+                .check(matches(isDisplayed()))
+                .check(matches(hasDescendant(withId(R.id.profile_header))));
+
+        intended(hasExtra("review", review));
+
+    }
+
+    // Test that error message is displayed after writing a comment exceeding MAX_REVIEW_LENGTH
+    @Test
+    public void checkErrorMessageWhenAddingAIncorrectLengthComment() {
+        editText.perform(typeText("a".repeat(MAX_REVIEW_LENGTH + 1)));
+        editText.perform(closeSoftKeyboard());
+
+        addItemButton.perform(click());
+        //activity.addItem();
+
+        errorText.check(matches(withText(
+                        String.format(Locale.ENGLISH, "Exceeded character limit: %d", MAX_REVIEW_LENGTH))));
+    }
+
+    // After the error message is displayed, it should disappears when user edits the comment to make it shorter
+    // It reappears if the length is still to long when adding the review
+    @Test
+    public void checkErrorMessageDisappearsWhenEditing() {
+        editText.perform(typeText("a".repeat(MAX_REVIEW_LENGTH + 1)));
+        editText.perform(closeSoftKeyboard());
+
+        addItemButton.perform(click());
+        //activity.addItem();
+
+        editText.perform(closeSoftKeyboard());
+        editText.perform(click(), closeSoftKeyboard());
```

Comment:
It is not clear what this code exactly does. Why is the keyboard closed twice ?

If addItemButton reopens the keyboard, it might be good to precise it through a small comment.